app/editor/src/features/content/form/ContentClipForm.tsx

```diff
@@ -0,0 +1,248 @@
+    try {
+      setItem(item);
+      toggle();
+    } catch {
+      // Ignore error a toast would have already been displayed with the error.
+    }
+  };
+
+  const downloadItem = async (item: IItemModel) => {
+    await storage.download(`${folder.path}/${item.name}`);
+  };
+
+  const attachItem = async (item: IItemModel) => {
+    await storage.attach(content.id, `${folder.path}/${item.name}`).then((data) => {
+      toast.success('Attachment added to this snippet.');
+    });
+  };
+
+  const selectItem = (item?: IItemModel) => {
+    setItem(item);
+    setCurrFile(!!item ? item.name : '');
+    setStreamUrl(
+      !!item ? `/api/editor/storage/stream?path=${folder.path}/${item.name}` : undefined,
+    );
+  };
+
+  const createClip = async () => {
+    if (!start || !end) {
+      toast.error('The clip start time and clip end time must both be set.');
+    } else if (parseInt(start) >= parseInt(end)) {
+      toast.error('The clip start time must be before the clip end time.');
+    } else {
+      setClipNbr(!!clipNbr ? clipNbr + 1 : 1);
+      await storage.clip(currFile, folder.path, start, end, clipNbr, prefix).then((data) => {
+        setFolder({ ...data, items: data.items });
+        setStart('');
+        setEnd('');
+      });
+    }
+  };
+
+  const joinClips = async () => {
+    await storage.join(currFile, folder.path, prefix).then((data) => {
+      setFolder({ ...data, items: data.items });
+      setClipNbr(1);
+      setStart('');
+      setEnd('');
+    });
+  };
+
+  const clipSubmit = async (event: any) => {
+    debugger;
+    var x = 1;
+  };
+
+  const setStartTime = () => {
+    setStart(!!videoRef.current ? videoRef.current?.currentTime.toFixed() : '');
+  };
+
+  const setEndTime = () => {
+    debugger;
+    setEnd(!!videoRef.current ? videoRef.current?.currentTime.toFixed() : '');
+  };
+
+  const navigate = (item?: IItemModel) => {
+    if (item?.isDirectory) {
+      setPath(`${folder.path}/${item?.name}`);
+    }
+  };
+
+  const navigateUp = () => {
+    var pathComps = folder.path.split('/');
+    if (pathComps.length === 3) {
+      setPath(pathComps[1]);
+    }
+  };
+
+  return (
+    <styled.ContentClipForm>
+      <div>
+        <Button className="navigate-up" onClick={() => navigateUp()} name="navigateUp">
+          Directory: {folder.path}
+        </Button>
+      </div>
+      <Row>
+        <Col>
+          <div className="file-table">
+            <ClipDirectoryTable
+              data={folder.items}
+              deleteItem={deleteItem}
+              selectItem={selectItem}
+              downloadItem={downloadItem}
+              attachItem={attachItem}
+              navigate={navigate}
+            />
+          </div>
+        </Col>
+      </Row>
+      <div className={!streamUrl ? 'hidden' : ''}>
+        <Row>
+          <div className="editing">Editing: {item?.name}</div>
+          <Col className="video" alignItems="stretch">
+            <video ref={videoRef} controls>
```

Comment:
Would be nice to make this a component if we're going to support all these types of sources.

Reply:
I would like to make this a different ticket as this is 1st Iteration.